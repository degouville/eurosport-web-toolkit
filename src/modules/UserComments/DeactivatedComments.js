import React from 'react';
import styled, { css } from 'react-emotion';
import { fontAlphaHeadline } from '../../typography';
import { medium } from '../../breakpoints';
import lockIcon from '../../assets/lock-icon.svg';
import { coreLightMinus1 } from '../../colors';

const StyledWrapper = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
  text-align: center;
  font-size: 12px;
  ${fontAlphaHeadline};
  color: ${coreLightMinus1};

  ${medium(css`
    font-size: 16px;
  `)};
`;

const StyledIcon = styled.img`
  margin-bottom: 20px;
  height: 46px;
`;

const DeactivatedComments = () => (
  <StyledWrapper>
    <StyledIcon src={lockIcon} alt="lock icon" />
    <div>Comments are deactivated for this publication</div>
  </StyledWrapper>
);

export default DeactivatedComments;
