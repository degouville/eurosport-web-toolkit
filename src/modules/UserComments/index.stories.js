/* eslint-disable no-underscore-dangle */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { select, boolean, number, object, text } from '@storybook/addon-knobs';
import { UserComments, AdInit, AdManager, AdPlacement } from '../..';
import activatedCommentsConfig from './mockData/activatedComments.json';
import deactivatedCommentsConfig from './mockData/deactivatedComments.json';

const userCommentsStories = storiesOf('Modules|UserComments', module);

const loginCallback = () => {
  const win = window.open('https://www.eurosport.fr/login.shtml', '_blank');
  win.focus();
};

const logoutCallback = () => {};

userCommentsStories.add('ActivatedUserComments', () => (
  <>
    <UserComments
      userToken={text('userToken', '')}
      loginCallback={loginCallback}
      logoutCallback={logoutCallback}
      livefyreConfig={object('livefyreConfig', activatedCommentsConfig)}
      nbCommentsText={text('nbCommentsText', '{{count}} comment')}
      nbCommentsTextPlural={text('nbCommentsTextPlural', '{{count}} comments')}
      deactivatedText={text('deactivatedText', 'Comments are deactivated for this publication')}
    />
  </>
));

userCommentsStories.add('DeactivatedUserComments', () => (
  <>
    <UserComments
      userToken={text('userToken', '')}
      loginCallback={loginCallback}
      logoutCallback={logoutCallback}
      livefyreConfig={object('livefyreConfig', deactivatedCommentsConfig)}
      nbCommentsText={text('nbCommentsText', '{{count}} comment')}
      nbCommentsTextPlural={text('nbCommentsTextPlural', '{{count}} comments')}
      deactivatedText={text('deactivatedText', 'Comments are deactivated for this publication')}
    />
  </>
));

userCommentsStories.add('ActivatedUserCommentsWithAds', () => (
  <>
    <AdInit
      domain="fr"
      pageType={select('pageType', Object.values(AdManager.enums.pageTypes), AdManager.enums.pageTypes.LIVE)}
      personalizeAds={boolean('personalizeAds', false)}
      advertisementSlotId={number('advertisementSlotId', 21725585473)}
      slotsConfig={object('slotsConfig', AdManager.defaultSlotsConfig)}
      keyValues={object('keyValues', { sport: 57 })}
    />

    <UserComments
      userToken={text('userToken', '')}
      loginCallback={loginCallback}
      logoutCallback={logoutCallback}
      livefyreConfig={object('livefyreConfig', activatedCommentsConfig)}
      topAdElement={<AdPlacement adType={AdManager.enums.adTypes.BANNER_SPONSORSHIP} />}
      rightAdElement={<AdPlacement adType={AdManager.enums.adTypes.MPU} isNoMobile isNoTablet />}
      nbCommentsText={text('nbCommentsText', '{{count}} comment')}
      nbCommentsTextPlural={text('nbCommentsTextPlural', '{{count}} comments')}
      deactivatedText={text('deactivatedText', 'Comments are deactivated for this publication')}
    />
  </>
));
