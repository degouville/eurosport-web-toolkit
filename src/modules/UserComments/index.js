import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import ActivatedComments, { ActivatedCommentsPropTypeShape } from './ActivatedComments';
import DeactivatedComments from './DeactivatedComments';

const StyledAdvertisement = styled.div`
  margin: 0 auto 30px;
  display: table;
`;

const UserComments = props => {
  const { livefyreConfig, topAdElement } = props;
  const comments = livefyreConfig.areCommentsActivated ? <ActivatedComments {...props} /> : <DeactivatedComments />;

  return (
    <>
      <StyledAdvertisement>{topAdElement}</StyledAdvertisement>
      {comments}
    </>
  );
};

UserComments.defaultProps = {
  userToken: '',
  topAdElement: null,
  rightAdElement: null,
};

UserComments.propTypes = {
  livefyreConfig: ActivatedCommentsPropTypeShape.isRequired,
  loginCallback: PropTypes.func.isRequired,
  logoutCallback: PropTypes.func.isRequired,
  userToken: PropTypes.string,
  topAdElement: PropTypes.element,
  rightAdElement: PropTypes.element,
};

export default UserComments;
