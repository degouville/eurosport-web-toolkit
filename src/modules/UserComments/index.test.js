import React from 'react';
import { shallow } from 'enzyme';
import UserComments from '.';
import DeactivatedComments from './DeactivatedComments';
import activatedCommentsConfig from './mockData/activatedComments.json';
import deactivatedCommentsConfig from './mockData/deactivatedComments.json';

describe('UserComments', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(
      <UserComments
        livefyreConfig={activatedCommentsConfig}
        loginCallback={() => null}
        logoutCallback={() => null}
        userToken=""
      />
    );
  });

  it('renders a UserComments Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders an activated UserComments Component', () => {
    expect(wrapper.find('ActivatedComments')).toHaveLength(1);
  });

  it('renders a deactivated UserComments Component', () => {
    wrapper.setProps({
      livefyreConfig: { deactivatedCommentsConfig },
    });
    expect(wrapper.find(DeactivatedComments)).toHaveLength(1);
  });
});
