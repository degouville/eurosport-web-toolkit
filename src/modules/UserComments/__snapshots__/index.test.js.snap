// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComments renders a UserComments Component 1`] = `
<Fragment>
  <Styled(div) />
  <ActivatedComments
    livefyreConfig={
      Object {
        "areCommentsActivated": true,
        "collection": Object {
          "articleId": "948848_4",
          "checksum": "5662654f95ebee3b55eec72f64b3a787",
          "collectionMeta": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1cmwiOiJodHRwczovL3d3dy5ldXJvc3BvcnQuZnIvZm9vdGJhbGwvbGlndWUtZGVzLWNoYW1waW9ucy8yMDE3LTIwMTgvbGl2ZS1yZWFsLW1hZHJpZC1wYXJpcy1zYWludC1nZXJtYWluX210Yzk0ODg0OC9saXZlLnNodG1sIiwidGl0bGUiOiJMaWd1ZSBkZXMgY2hhbXBpb25zIDIwMTctMjAxOCBSZWFsIE1hZHJpZC1QYXJpcyBTYWludC1HZXJtYWluIiwidHlwZSI6ImxpdmVjb21tZW50cyIsImFydGljbGVJZCI6Ijk0ODg0OF80In0.QlshF5MD-J1ivClFm6GPp_X77z6njW-cvhLW2Lv3fLk",
        },
        "config": Object {
          "networkDomain": "eurosport-fr.fyre.co",
          "scriptUrl": "//cdn.livefyre.com/Livefyre.js",
          "siteId": 373116,
        },
        "labels": Object {
          "MultipleMediaNotAllowedError": "Your permissions only grant you one media attachment at a time.",
          "archiveHeaderTitle": "From the Archive",
          "archiveShowMore": "Show More",
          "backToComments": "Back to Newest",
          "backToHotThreads": "Back to Hot Threads",
          "banCancelButton": "Cancel",
          "banConfirmButton": "OK",
          "banConfirmation": "Are you sure you want to ban this user?",
          "banTitle": "Ban user",
          "banUserButton": "Ban user",
          "banUserTooltip": "Ban user",
          "bozoButton": "Bozo",
          "bozoCommentTooltip": "Bozo",
          "commentCountLabel": "%s comment",
          "commentCountLabelPlural": "%s comments",
          "commentNotifier": "New Comment",
          "commentNotifierPlural": "New Comments",
          "commentTombstone": "This comment has been deleted",
          "customHtmlDialogTitle": "Add Custom HTML",
          "deleteButton": "Delete",
          "deleteCommentTooltip": "Delete",
          "displayName": "%s",
          "displayNameGuest": "%s (guest)",
          "editButton": "Edit",
          "editProfile": "Edit profile",
          "editorDisabled": "This conversation is currently closed to new comments.",
          "errorAuthError": "You're not authorized to post a comment on this conversation",
          "errorCommentsNotAllowed": "Comments are not allowed on this conversation",
          "errorDefault": "An error has occurred. Please try again.",
          "errorDuplicate": "As much as you liked your comment, you're not allowed to post it twice.",
          "errorEditDuplicate": "You must change the body of the comment when you edit it.",
          "errorEditNotAllowed": "You're not allowed to edit comments on this conversation.",
          "errorEditTimeExceeded": "Sorry, your comment editing period has expired.",
          "errorEmpty": "It seems you're attempting to post an empty comment.",
          "errorExpired": "Your session has expired. Please reload the page.",
          "errorFlagNotSelected": "Please select a flag type.",
          "errorGuestLiked": "Sorry, only those with accounts can like content.",
          "errorInsufficientPermissions": "Insufficient Permissions",
          "errorInvalidChar": "It seems you're attempting to post an invalid character.",
          "errorLikeOwnComment": "You cannot like your own comment",
          "errorMalformed": "It seems you're attempting to post malformed content.",
          "errorMaxChars": "Sorry, your comment is too long. Please edit and try again.",
          "errorMediaNotAvailable": "Media is no longer visible.",
          "errorShowMore": "There was an error loading more comments.",
          "featureButton": "Feature",
          "featureCommentTooltip": "Feature",
          "featuredCommentTooltip": "Unfeature",
          "featuredCommentsTag": "Featured",
          "featuredCommentsTitlePlural": "Featured Comment",
          "flagButton": "Flag",
          "flagCancelButton": "Cancel",
          "flagCommentTooltip": "...More",
          "flagConfirmButton": "OK",
          "flagConfirmationMessage": "Flag %s's comment as %s?",
          "flagDefaultSelectOption": "Select",
          "flagDisagree": "Disagree",
          "flagDisagreeButton": "Disagree",
          "flagDisagreeCommentTooltip": "Disagree",
          "flagEmail": "Email",
          "flagEmailPlaceholder": "Your e-mail address",
          "flagNotes": "Notes",
          "flagNotesPlaceholder": "Notes",
          "flagOffTopic": "Off-Topic",
          "flagOffTopicButton": "Off-Topic",
          "flagOffTopicCommentTooltip": "Off-Topic",
          "flagOffensive": "Offensive",
          "flagOffensiveButton": "Offensive",
          "flagOffensiveCommentTooltip": "Offensive",
          "flagSpam": "Spam",
          "flagSpamButton": "Spam",
          "flagSpamCommentTooltip": "Spam",
          "flagSubtitle": "Flag as",
          "flagSuccessMsg": "Comment has been flagged.",
          "flagTitle": "Flag %s's comment",
          "follow": "+ Follow",
          "hotThreadsContentNotFoundMsg": "There are not enough threads yet.",
          "hovercardViewProfile": "View Full Profile",
          "likeButton": "Like",
          "likesPlural": "Likes",
          "likesSingular": "Like",
          "listenerCount": "person listening",
          "listenerCountPlural": "people listening",
          "liveblogFollow": "Follow Live Blog",
          "liveblogNotifier": "New Post",
          "liveblogNotifierPlural": "New Posts",
          "liveblogPostCountLabel": "post",
          "liveblogPostCountLabelPlural": "posts",
          "liveblogUnfollow": "Unfollow Live Blog",
          "livechatOtherInstruction": "Press command+enter to post",
          "livechatPostButtonLabel": "Post",
          "livechatPostEditButton": "Edit",
          "livechatWindowsInstruction": "Press control+enter to post",
          "loadingPermalink": "Loading short url...",
          "mentionCancelButton": "Cancel",
          "mentionConfirmButton": "OK",
          "mentionDefaultText": "I mentioned you in a Livefyre comment!",
          "mentionErrorGeneral": "Oops! Something went wrong! Livefyre has been alerted.",
          "mentionErrorNoneSelected": "You must have at least one mention enabled.",
          "mentionMenuTitle": "To see and mention your friends",
          "mentionSubtitleTwitter": "Share Tweet to:",
          "mentionSuccessMsg": "Mentions have been successfully sent.",
          "mentionTitle": "Share Mention",
          "mentionTwitterConnect": "Connect to Twitter",
          "mentionTwitterFetching": "Fetching Friends...",
          "modQ": "ModQ",
          "moderator": "moderator",
          "moderatorEditTimestamp": "Edited by a Moderator",
          "moreButton": "...More",
          "newComment": "New Comment",
          "newComments": "New Comments",
          "newReplies": "new replies",
          "newReply": "new reply",
          "notificationSettings": "Notification settings",
          "pendingComment": "pending",
          "permalinkNotFoundMsg": "This comment is no longer visible.",
          "postAsButton": "Post comment as…",
          "postButton": "Post comment",
          "postEditAsButton": "Edit comment as...",
          "postEditButton": "Edit comment",
          "postEditCancelButton": "Cancel",
          "postReplyAsButton": "Post comment as…",
          "postReplyButton": "Post comment",
          "quickProfileTooltip": "Quick Profile",
          "refresh": "Refresh",
          "replyButton": "Reply",
          "replyButtonPlural": "Replies",
          "replyButtonSingular": "reply",
          "shareButton": "Share",
          "shareButtonText": "Share",
          "shareCommentTooltip": "Share",
          "shareDefaultText": "Check out %s's post!",
          "shareLabel": "Share on:",
          "sharePermalink": "Permalink",
          "shareText": "I just posted a comment. Check it out!",
          "shareTextFacebook": "blank",
          "shareTextLinkedin": "blank",
          "shareTextTwitter": "blank",
          "shareTitle": "Share Comment",
          "showMore": "Show More Comments",
          "showMoreLiveblog": "Show More Posts",
          "signIn": "Sign in",
          "signInGuest": "Sign in or Post as Guest",
          "signOut": "Sign out",
          "siteAdmin": "Admin Console",
          "sortHotThreads": "Hot Threads",
          "sortNewest": "Newest",
          "sortOldest": "Oldest",
          "sortTopComments": "Top Comments",
          "streamRefreshMsg": "See what's new.",
          "streamSorting": "Loading",
          "threadBreakoutButton": "Show entire Thread",
          "timeDaysAgo": "day ago",
          "timeDaysAgoPlural": "days ago",
          "timeHoursAgo": "hour ago",
          "timeHoursAgoPlural": "hours ago",
          "timeInDays": "in %s day",
          "timeInDaysPlural": "in %s days",
          "timeInHours": "in %s hour",
          "timeInHoursPlural": "in %s hours",
          "timeInMinutes": "in %s minute",
          "timeInMinutesPlural": "in %s minutes",
          "timeJustNow": "just now",
          "timeMinutesAgo": "minute ago",
          "timeMinutesAgoPlural": "minutes ago",
          "toggleCollapse": "Toggle Collapse",
          "topCommentsContentNotFoundMsg": "There are not enough likes yet.",
          "unfeatureButton": "Unfeature",
          "unfollow": "- Unfollow",
          "unlikeButton": "Unlike",
          "voteDownvote": "No",
          "voteReplyTitle": "Was this reply helpful?",
          "voteTitle": "Was this comment helpful?",
          "voteUpvote": "Yes",
        },
      }
    }
    loginCallback={[Function]}
    logoutCallback={[Function]}
    rightAdElement={null}
    topAdElement={null}
    userToken=""
  />
</Fragment>
`;
